src/wp-includes/post-thumbnail-template.php

```diff
@@ -113,6 +113,17 @@ function update_post_thumbnail_cache( $wp_query = null ) {
 	$thumb_ids = array();
```

Comment:
This avoids repeated calls to the database 

```suggestion
	$thumb_ids = array();

	/*
	 * $wp_query may contain an array of post objects or post IDs.
	 *
	 * This ensures the cache is primed for all post objects to avoid
	 * repeated `get_post()` calls in `get_the_post_thumbnail()`.
	 */
	$post_ids = array();
	foreach ( $wp_query->posts as $post ) {
		if ( $post instanceof WP_Post ) {
			$post_ids[] = $post->ID;
		} elseif ( is_int( $post ) ) {
			$post_ids[] = $post;
		}
	}
	_prime_post_caches( $post_ids, false, true );
```